code/modules/antagonists/thief/thief_objectives.dm

```diff
@@ -0,0 +1,111 @@
+))
+
+/datum/objective/hoarder
+	name = "hoarding"
+	explanation_text = "Hoard as many items as you can in one area!"
+	///what item we want to get many of
+	var/atom/movable/target_type
+	///how many we want for greentext
+	var/amount = 7
+	///and where do we want it roundend (this is a type, not a ref)
+	var/turf/hoarder_turf
+
+/datum/objective/hoarder/find_target(dupe_search_range, blacklist)
+	amount = rand(amount - 2, amount + 2)
+	target_type = pick(GLOB.hoarder_targets)
+	add_action()
+
+/datum/objective/hoarder/proc/add_action()
+	var/list/owners = get_owners()
+	var/datum/mind/hoardpicker = owners[1]
+	var/datum/action/declare_hoard/declare = new /datum/action/declare_hoard(hoardpicker.current)
+	declare.weak_objective = WEAKREF(src)
+	declare.Grant(hoardpicker.current)
+
+/datum/objective/hoarder/update_explanation_text()
+	var/obj/item/target_item = target_type
+	explanation_text = "Hoard as many [initial(target_item.name)]\s as you can in maintenance (after declaring a spot)! At least [amount] will do."
+
+/datum/objective/hoarder/check_completion()
+	. = ..()
+	if(.)
+		return TRUE
+	var/stolen_amount = 0
+	if(!hoarder_turf)
+		return FALSE //they never set up their hoard spot, so they couldn't have done their objective
+	for(var/atom/movable/in_target_turf in hoarder_turf.get_all_contents())
+		if(istype(in_target_turf, target_type))
+			if(!valid_target(in_target_turf))
+				continue
+			stolen_amount++
+	return stolen_amount >= amount
+
+/datum/objective/hoarder/proc/valid_target(atom/movable/target)
+	return TRUE
+
+///for the deranged flavor
+/datum/objective/hoarder/bodies
+	name = "corpse hoarding"
+	explanation_text = "Hoard as many dead bodies as you can in one area!"
+	amount = 5 //little less, bodies are hard when you can't kill like antags
+
+/datum/objective/hoarder/bodies/find_target(dupe_search_range, blacklist)
+	amount = rand(amount - 2, amount + 2)
+	target_type = /mob/living/carbon/human
+	add_action()
+
+/datum/objective/hoarder/bodies/valid_target(mob/living/carbon/human/target)
+	if(target.stat != DEAD)
+		return FALSE
+	return TRUE
+
+/datum/objective/hoarder/bodies/update_explanation_text()
+	explanation_text = "Hoard as many dead bodies as you can in maintenance (after declaring a spot)! At least [amount] will do."
+
+/datum/action/declare_hoard
+	name = "Declare Hoard"
+	desc = "Declare a new hoarding spot on the ground you're standing on. Items on this floor will count for your objective."
+	icon_icon = 'icons/mob/actions/actions_minor_antag.dmi'
+	button_icon_state = "hoard"
+	///weak reference to the objective this action targets, set by hoarder objective
+	var/datum/weakref/weak_objective
+
+/datum/action/declare_hoard/Trigger(trigger_flags)
+	if(owner.incapacitated())
+		owner.balloon_alert(owner, "not while incapacitated!")
+		return
+	var/area/owner_area = get_area(owner)
+	if(!(owner_area.type in typesof(/area/maintenance)))
+		owner.balloon_alert(owner, "hoard must be in maintenance!")
+		return
```

Comment:
this maintenance only requirement kind of fucks you over if you're a job that doesn't start with maint access

like, yeah, you can make a false wall in the chapel, I guess? but it really feels like you're being forced to be out of your element instead of maintenance being where you belong.

imo, the better way to do this would be to let you designate any area for this, with some blacklists based on what you're hoarding (i.e. no hoarding bodies in the morgue, insuls in engineering, batons in sec, etc.).